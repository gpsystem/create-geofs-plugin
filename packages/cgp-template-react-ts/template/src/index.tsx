/// <reference types="jquery" />
/// <reference types="@geps/geofs-types" />

import React from "react";
import { createRoot } from "react-dom/client";
import { App } from "./App";

// Replace the selector with a link to your UI host!
// Example: #geofs-ui-right
// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
const root = createRoot(document.querySelector("REPLACE ME")!);
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
