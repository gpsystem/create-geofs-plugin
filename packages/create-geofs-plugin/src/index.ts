import { join } from "node:path";
import { dir, DirectoryResult } from "tmp-promise";
import { Configuration, parseConfig } from "./config";
import { copyDir } from "./fsHelpers";
import expandTemplateJson, { ExpandedTemplateJson } from "./template";

export async function start(argv: string[]): Promise<void> {
  const config: Configuration = parseConfig(argv);
  const { path: tmpDirName, cleanup: cleanupTempDir }: DirectoryResult =
    await dir({
      unsafeCleanup: true,
    });

  copyDir(config.templateDir, tmpDirName, { overwrite: config.overwrite });
  const { dependencies: templateDeps, eslintConfig }: ExpandedTemplateJson =
    expandTemplateJson(join(tmpDirName, "template.json"));
  // remove everything that starts with __
  // copy everything to the target dir
  // install the dependencies
  await cleanupTempDir();
}
