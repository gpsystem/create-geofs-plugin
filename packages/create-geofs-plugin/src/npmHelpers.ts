import { execSync } from "node:child_process";
import { join } from "node:path";
import { outputFile } from "./fsHelpers";

export function initializeNpm(
  targetDir: string,
  templateDeps: string[],
  ci: boolean
): void {
  const firstPackageJson: Record<string, unknown> = {
    name: "geofs-plugin",
    version: "",
    private: true,
  };

  outputFile(
    join(targetDir, "package.json"),
    JSON.stringify(firstPackageJson, null, 2),
    {
      force: true,
    }
  );
  if (!ci) execSync(`npm i ${templateDeps.join(" ")} -D`, { cwd: targetDir });
}
