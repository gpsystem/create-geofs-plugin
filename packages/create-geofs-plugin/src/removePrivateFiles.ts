export default function removePrivateFiles(directoryPath: string) {
  // all private files start with __
}
