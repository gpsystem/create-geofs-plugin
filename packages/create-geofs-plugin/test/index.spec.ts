import { readdirSync, readFileSync } from "node:fs";
import { join, relative, sep } from "node:path";
import { start } from "../src/index";
import {
  allArgPermutations,
  createArgsForProgram,
  createTestDirectory__RENAME_LATER,
  TestDirectoryResults,
} from "./utils/index";

type StringObject = { [key: string]: string };

function createDirectoryTree(
  startDirectory: string,
  { recursing = false }: { recursing?: boolean } = {}
): StringObject {
  const files: StringObject = {};

  for (const file of readdirSync(startDirectory, { withFileTypes: true })) {
    const filePath: string = join(startDirectory, file.name);

    if (file.isDirectory()) {
      const subdirectoryTree: StringObject = createDirectoryTree(filePath, {
        recursing: true,
      });

      // reviewers, mark this to resolve the name conflict
      for (const [filePath, fileContents] of Object.entries(subdirectoryTree)) {
        files[filePath] = fileContents;
      }
    } else if (file.isFile()) {
      files[filePath] = readFileSync(filePath, "utf8");
    }
  }

  // make all the names in files relative to the start directory
  if (!recursing)
    return Object.entries(files).reduce<StringObject>(
      (acc, [filePath, fileContents]) => {
        acc[relative(startDirectory, filePath).split(sep).join("/")] =
          fileContents;
        return acc;
      },
      {}
    );
  else return files;
}

describe("integration tests", () => {
  const { cleanup, directoryPath }: TestDirectoryResults =
    createTestDirectory__RENAME_LATER();

  afterEach(() => cleanup());

  for (const [name, args] of allArgPermutations) {
    test(`integration test ${name}`, async () => {
      await expect(
        start(createArgsForProgram(args, directoryPath), true)
      ).resolves.toBeUndefined();

      expect(createDirectoryTree(directoryPath)).toMatchSnapshot();
    });
  }
});
