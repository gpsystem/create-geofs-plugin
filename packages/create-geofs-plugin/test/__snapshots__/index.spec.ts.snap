// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration tests integration test overwrite:false 1`] = `
Object {
  ".eslintrc.yml": "root: true
env:
  browser: true
  es2021: true
  node: true
extends:
  - eslint:recommended
  - plugin:import/recommended
  - prettier
globals:
  PAGE_PATH: readonly
  geofs: readonly
  ui: readonly
  flight: readonly
  objects: readonly
  controls: readonly
  weather: readonly
  audio: readonly
  multiplayer: readonly
  instruments: readonly
  absMin: readonly
  span: readonly
  boundHours24: readonly
  fixAngle: readonly
  fixAngle360: readonly
  fixAngles360: readonly
  fixAngles: readonly
  exponentialSmoothingV3: readonly
  exponentialSmoothing: readonly
  getBuildingCollision: readonly
  xyz2lla: readonly
  xy2ll: readonly
  lla2xyz: readonly
  ll2xy: readonly
  clamp: readonly
  geoDecodeLocation: readonly
  lookAt: readonly
  getURLParameters: readonly
  clone: readonly
  PID: readonly
  Overlay: readonly
  rigidBody: readonly
  Object3D: readonly
  Indicator: readonly
  GlassPanel: readonly
parserOptions:
  ecmaVersion: latest
  sourceType: module
rules:
  eqeqeq:
    - error
  import/order:
    - error
    - alphabetize:
        order: asc
        caseInsensitive: true
",
  "package.json": "{
  \\"name\\": \\"geofs-plugin\\",
  \\"version\\": \\"\\",
  \\"private\\": true
}",
  "src/index.js": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

// start writing your code here!
// you can reference the geofs apis in intellisense because of the triple-slash directives
// the apis from geofs plugin system are not available yet
",
  "webpack.config.js": "const { join } = require(\\"node:path\\");

/** @type {import(\\"webpack\\").Configuration} */
const config = {
  entry: join(__dirname, \\"src\\", \\"index.js\\"),
  mode: \\"production\\",
  module: {
    rules: [
      {
        test: /\\\\.m?js$/,
        exclude: /node_modules/,
        use: {
          loader: \\"babel-loader\\",
          options: {
            presets: [\\"@babel/preset-env\\"],
          },
        },
      },
    ],
  },
  output: {
    clean: true,
    filename: \\"index.js\\",
    path: join(__dirname, \\"dist\\"),
  },
};

module.exports = config;
",
}
`;

exports[`integration tests integration test overwrite:false/template:react 1`] = `
Object {
  ".eslintrc.yml": "root: true
env:
  browser: true
  es2021: true
  node: true
extends:
  - plugin:react/recommended
  - plugin:import/typescript
  - prettier
parserOptions:
  ecmaVersion: latest
  sourceType: module
  ecmaFeatures:
    jsx: true
rules:
  eqeqeq:
    - error
  import/order:
    - error
    - alphabetize:
        order: asc
        caseInsensitive: true
parser: '@typescript-eslint/parser'
plugins:
  - react
overrides:
  - files:
      - '**/*.js'
    rules:
      '@typescript-eslint/no-var-requires':
        - 'off'
settings:
  import/parsers:
    '@typescript-eslint/parser':
      - .ts
      - .tsx
  react:
    version: detect
",
  "package.json": "{
  \\"name\\": \\"geofs-plugin\\",
  \\"version\\": \\"\\",
  \\"private\\": true
}",
  "src/App.js": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

import React from \\"react\\";

export function App() {
  // start writing your code here!
  // you can reference the geofs apis in intellisense because of the triple-slash directives
  // the apis from geofs plugin system are not available yet

  return <h1>Test</h1>;
}
",
  "src/index.js": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

import React from \\"react\\";
import { createRoot } from \\"react-dom/client\\";
import { App } from \\"./App\\";

// Replace the selector with a link to your UI host!
// Example: #geofs-ui-right
const root = createRoot(document.querySelector(\\"REPLACE ME\\"));
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
",
  "webpack.config.js": "const { join } = require(\\"node:path\\");

/** @type {import(\\"webpack\\").Configuration} */
const config = {
  entry: join(__dirname, \\"src\\", \\"index.js\\"),
  mode: \\"production\\",
  module: {
    rules: [
      {
        test: /\\\\.m?js$/,
        exclude: /node_modules/,
        use: {
          loader: \\"babel-loader\\",
          options: {
            presets: [\\"@babel/preset-env\\", \\"@babel/preset-react\\"],
          },
        },
      },
    ],
  },
  output: {
    clean: true,
    filename: \\"index.js\\",
    path: join(__dirname, \\"dist\\"),
  },
};

module.exports = config;
",
}
`;

exports[`integration tests integration test overwrite:false/template:react-ts 1`] = `
Object {
  ".eslintrc.yml": "root: true
env:
  browser: true
  es2021: true
  node: true
extends:
  - plugin:react/recommended
  - plugin:import/typescript
  - prettier
parserOptions:
  ecmaVersion: latest
  sourceType: module
  ecmaFeatures:
    jsx: true
rules:
  eqeqeq:
    - error
  import/order:
    - error
    - alphabetize:
        order: asc
        caseInsensitive: true
parser: '@typescript-eslint/parser'
plugins:
  - react
overrides:
  - files:
      - '**/*.js'
    rules:
      '@typescript-eslint/no-var-requires':
        - 'off'
settings:
  import/parsers:
    '@typescript-eslint/parser':
      - .ts
      - .tsx
  react:
    version: detect
",
  "package.json": "{
  \\"name\\": \\"geofs-plugin\\",
  \\"version\\": \\"\\",
  \\"private\\": true
}",
  "src/App.tsx": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

import React from \\"react\\";

export function App() {
  // start writing your code here!
  // you can reference the geofs apis in intellisense because of the triple-slash directives
  // the apis from geofs plugin system are not available yet

  return <h1>Test</h1>;
}
",
  "src/index.tsx": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

import React from \\"react\\";
import { createRoot } from \\"react-dom/client\\";
import { App } from \\"./App\\";

// Replace the selector with a link to your UI host!
// Example: #geofs-ui-right
// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
const root = createRoot(document.querySelector(\\"REPLACE ME\\")!);
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
",
  "src/tsconfig.json": "{
  \\"compilerOptions\\": {
    \\"target\\": \\"es2015\\",
    \\"module\\": \\"esnext\\",
    \\"esModuleInterop\\": true,
    \\"moduleResolution\\": \\"node\\",
    \\"resolveJsonModule\\": true,
    \\"forceConsistentCasingInFileNames\\": true,
    \\"strict\\": true,
    \\"jsx\\": \\"react\\",
    \\"skipLibCheck\\": true
  },
  \\"include\\": [\\"**/*\\"]
}
",
  "webpack.config.js": "// @ts-check
const { join } = require(\\"node:path\\");

/** @type {import(\\"webpack\\").Configuration} */
const config = {
  entry: join(__dirname, \\"src\\", \\"index.js\\"),
  mode: \\"production\\",
  module: {
    rules: [
      {
        test: /\\\\.tsx?$/,
        exclude: /(node_modules)/,
        use: [
          {
            loader: \\"ts-loader\\",
          },
        ],
      },
    ],
  },
  output: {
    clean: true,
    filename: \\"index.js\\",
    path: join(__dirname, \\"dist\\"),
  },
};

module.exports = config;
",
}
`;

exports[`integration tests integration test overwrite:false/template:typescript 1`] = `
Object {
  ".eslintrc.yml": "root: true
env:
  browser: true
  es2021: true
  node: true
extends:
  - plugin:@typescript-eslint/recommended
  - plugin:import/typescript
  - prettier
parserOptions:
  ecmaVersion: latest
  sourceType: module
rules:
  eqeqeq:
    - error
  import/order:
    - error
    - alphabetize:
        order: asc
        caseInsensitive: true
parser: '@typescript-eslint/parser'
plugins:
  - '@typescript-eslint'
overrides:
  - files:
      - '**/*.js'
    rules:
      '@typescript-eslint/no-var-requires':
        - 'off'
settings:
  import/parsers:
    '@typescript-eslint/parser':
      - .ts
      - .tsx
",
  "package.json": "{
  \\"name\\": \\"geofs-plugin\\",
  \\"version\\": \\"\\",
  \\"private\\": true
}",
  "src/index.ts": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

// start writing your code here!
// you can reference the geofs apis in intellisense because of the triple-slash directives
// the apis from geofs plugin system are not available yet
",
  "tsconfig.json": "{
  \\"compilerOptions\\": {
    \\"target\\": \\"es2015\\",
    \\"module\\": \\"esnext\\",
    \\"esModuleInterop\\": true,
    \\"moduleResolution\\": \\"node\\",
    \\"resolveJsonModule\\": true,
    \\"forceConsistentCasingInFileNames\\": true,
    \\"strict\\": true,
    \\"skipLibCheck\\": true
  },
  \\"include\\": [\\"**/*\\"]
}
",
  "webpack.config.js": "const { join } = require(\\"node:path\\");

/** @type {import(\\"webpack\\").Configuration} */
const config = {
  entry: join(__dirname, \\"src\\", \\"index.ts\\"),
  mode: \\"production\\",
  module: {
    rules: [
      {
        test: /\\\\.tsx?$/,
        exclude: /(node_modules)/,
        use: [
          {
            loader: \\"ts-loader\\",
          },
        ],
      },
    ],
  },
  resolve: {
    extensions: [\\".tsx\\", \\".ts\\", \\".js\\"],
  },
  output: {
    clean: true,
    filename: \\"index.js\\",
    path: join(__dirname, \\"dist\\"),
  },
};

module.exports = config;
",
}
`;

exports[`integration tests integration test overwrite:true 1`] = `
Object {
  ".eslintrc.yml": "root: true
env:
  browser: true
  es2021: true
  node: true
extends:
  - eslint:recommended
  - plugin:import/recommended
  - prettier
globals:
  PAGE_PATH: readonly
  geofs: readonly
  ui: readonly
  flight: readonly
  objects: readonly
  controls: readonly
  weather: readonly
  audio: readonly
  multiplayer: readonly
  instruments: readonly
  absMin: readonly
  span: readonly
  boundHours24: readonly
  fixAngle: readonly
  fixAngle360: readonly
  fixAngles360: readonly
  fixAngles: readonly
  exponentialSmoothingV3: readonly
  exponentialSmoothing: readonly
  getBuildingCollision: readonly
  xyz2lla: readonly
  xy2ll: readonly
  lla2xyz: readonly
  ll2xy: readonly
  clamp: readonly
  geoDecodeLocation: readonly
  lookAt: readonly
  getURLParameters: readonly
  clone: readonly
  PID: readonly
  Overlay: readonly
  rigidBody: readonly
  Object3D: readonly
  Indicator: readonly
  GlassPanel: readonly
parserOptions:
  ecmaVersion: latest
  sourceType: module
rules:
  eqeqeq:
    - error
  import/order:
    - error
    - alphabetize:
        order: asc
        caseInsensitive: true
",
  "package.json": "{
  \\"name\\": \\"geofs-plugin\\",
  \\"version\\": \\"\\",
  \\"private\\": true
}",
  "src/index.js": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

// start writing your code here!
// you can reference the geofs apis in intellisense because of the triple-slash directives
// the apis from geofs plugin system are not available yet
",
  "webpack.config.js": "const { join } = require(\\"node:path\\");

/** @type {import(\\"webpack\\").Configuration} */
const config = {
  entry: join(__dirname, \\"src\\", \\"index.js\\"),
  mode: \\"production\\",
  module: {
    rules: [
      {
        test: /\\\\.m?js$/,
        exclude: /node_modules/,
        use: {
          loader: \\"babel-loader\\",
          options: {
            presets: [\\"@babel/preset-env\\"],
          },
        },
      },
    ],
  },
  output: {
    clean: true,
    filename: \\"index.js\\",
    path: join(__dirname, \\"dist\\"),
  },
};

module.exports = config;
",
}
`;

exports[`integration tests integration test overwrite:true/template:react 1`] = `
Object {
  ".eslintrc.yml": "root: true
env:
  browser: true
  es2021: true
  node: true
extends:
  - plugin:react/recommended
  - plugin:import/typescript
  - prettier
parserOptions:
  ecmaVersion: latest
  sourceType: module
  ecmaFeatures:
    jsx: true
rules:
  eqeqeq:
    - error
  import/order:
    - error
    - alphabetize:
        order: asc
        caseInsensitive: true
parser: '@typescript-eslint/parser'
plugins:
  - react
overrides:
  - files:
      - '**/*.js'
    rules:
      '@typescript-eslint/no-var-requires':
        - 'off'
settings:
  import/parsers:
    '@typescript-eslint/parser':
      - .ts
      - .tsx
  react:
    version: detect
",
  "package.json": "{
  \\"name\\": \\"geofs-plugin\\",
  \\"version\\": \\"\\",
  \\"private\\": true
}",
  "src/App.js": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

import React from \\"react\\";

export function App() {
  // start writing your code here!
  // you can reference the geofs apis in intellisense because of the triple-slash directives
  // the apis from geofs plugin system are not available yet

  return <h1>Test</h1>;
}
",
  "src/index.js": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

import React from \\"react\\";
import { createRoot } from \\"react-dom/client\\";
import { App } from \\"./App\\";

// Replace the selector with a link to your UI host!
// Example: #geofs-ui-right
const root = createRoot(document.querySelector(\\"REPLACE ME\\"));
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
",
  "webpack.config.js": "const { join } = require(\\"node:path\\");

/** @type {import(\\"webpack\\").Configuration} */
const config = {
  entry: join(__dirname, \\"src\\", \\"index.js\\"),
  mode: \\"production\\",
  module: {
    rules: [
      {
        test: /\\\\.m?js$/,
        exclude: /node_modules/,
        use: {
          loader: \\"babel-loader\\",
          options: {
            presets: [\\"@babel/preset-env\\", \\"@babel/preset-react\\"],
          },
        },
      },
    ],
  },
  output: {
    clean: true,
    filename: \\"index.js\\",
    path: join(__dirname, \\"dist\\"),
  },
};

module.exports = config;
",
}
`;

exports[`integration tests integration test overwrite:true/template:react-ts 1`] = `
Object {
  ".eslintrc.yml": "root: true
env:
  browser: true
  es2021: true
  node: true
extends:
  - plugin:react/recommended
  - plugin:import/typescript
  - prettier
parserOptions:
  ecmaVersion: latest
  sourceType: module
  ecmaFeatures:
    jsx: true
rules:
  eqeqeq:
    - error
  import/order:
    - error
    - alphabetize:
        order: asc
        caseInsensitive: true
parser: '@typescript-eslint/parser'
plugins:
  - react
overrides:
  - files:
      - '**/*.js'
    rules:
      '@typescript-eslint/no-var-requires':
        - 'off'
settings:
  import/parsers:
    '@typescript-eslint/parser':
      - .ts
      - .tsx
  react:
    version: detect
",
  "package.json": "{
  \\"name\\": \\"geofs-plugin\\",
  \\"version\\": \\"\\",
  \\"private\\": true
}",
  "src/App.tsx": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

import React from \\"react\\";

export function App() {
  // start writing your code here!
  // you can reference the geofs apis in intellisense because of the triple-slash directives
  // the apis from geofs plugin system are not available yet

  return <h1>Test</h1>;
}
",
  "src/index.tsx": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

import React from \\"react\\";
import { createRoot } from \\"react-dom/client\\";
import { App } from \\"./App\\";

// Replace the selector with a link to your UI host!
// Example: #geofs-ui-right
// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
const root = createRoot(document.querySelector(\\"REPLACE ME\\")!);
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
",
  "src/tsconfig.json": "{
  \\"compilerOptions\\": {
    \\"target\\": \\"es2015\\",
    \\"module\\": \\"esnext\\",
    \\"esModuleInterop\\": true,
    \\"moduleResolution\\": \\"node\\",
    \\"resolveJsonModule\\": true,
    \\"forceConsistentCasingInFileNames\\": true,
    \\"strict\\": true,
    \\"jsx\\": \\"react\\",
    \\"skipLibCheck\\": true
  },
  \\"include\\": [\\"**/*\\"]
}
",
  "webpack.config.js": "// @ts-check
const { join } = require(\\"node:path\\");

/** @type {import(\\"webpack\\").Configuration} */
const config = {
  entry: join(__dirname, \\"src\\", \\"index.js\\"),
  mode: \\"production\\",
  module: {
    rules: [
      {
        test: /\\\\.tsx?$/,
        exclude: /(node_modules)/,
        use: [
          {
            loader: \\"ts-loader\\",
          },
        ],
      },
    ],
  },
  output: {
    clean: true,
    filename: \\"index.js\\",
    path: join(__dirname, \\"dist\\"),
  },
};

module.exports = config;
",
}
`;

exports[`integration tests integration test overwrite:true/template:typescript 1`] = `
Object {
  ".eslintrc.yml": "root: true
env:
  browser: true
  es2021: true
  node: true
extends:
  - plugin:@typescript-eslint/recommended
  - plugin:import/typescript
  - prettier
parserOptions:
  ecmaVersion: latest
  sourceType: module
rules:
  eqeqeq:
    - error
  import/order:
    - error
    - alphabetize:
        order: asc
        caseInsensitive: true
parser: '@typescript-eslint/parser'
plugins:
  - '@typescript-eslint'
overrides:
  - files:
      - '**/*.js'
    rules:
      '@typescript-eslint/no-var-requires':
        - 'off'
settings:
  import/parsers:
    '@typescript-eslint/parser':
      - .ts
      - .tsx
",
  "package.json": "{
  \\"name\\": \\"geofs-plugin\\",
  \\"version\\": \\"\\",
  \\"private\\": true
}",
  "src/index.ts": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

// start writing your code here!
// you can reference the geofs apis in intellisense because of the triple-slash directives
// the apis from geofs plugin system are not available yet
",
  "tsconfig.json": "{
  \\"compilerOptions\\": {
    \\"target\\": \\"es2015\\",
    \\"module\\": \\"esnext\\",
    \\"esModuleInterop\\": true,
    \\"moduleResolution\\": \\"node\\",
    \\"resolveJsonModule\\": true,
    \\"forceConsistentCasingInFileNames\\": true,
    \\"strict\\": true,
    \\"skipLibCheck\\": true
  },
  \\"include\\": [\\"**/*\\"]
}
",
  "webpack.config.js": "const { join } = require(\\"node:path\\");

/** @type {import(\\"webpack\\").Configuration} */
const config = {
  entry: join(__dirname, \\"src\\", \\"index.ts\\"),
  mode: \\"production\\",
  module: {
    rules: [
      {
        test: /\\\\.tsx?$/,
        exclude: /(node_modules)/,
        use: [
          {
            loader: \\"ts-loader\\",
          },
        ],
      },
    ],
  },
  resolve: {
    extensions: [\\".tsx\\", \\".ts\\", \\".js\\"],
  },
  output: {
    clean: true,
    filename: \\"index.js\\",
    path: join(__dirname, \\"dist\\"),
  },
};

module.exports = config;
",
}
`;

exports[`integration tests integration test template:react 1`] = `
Object {
  ".eslintrc.yml": "root: true
env:
  browser: true
  es2021: true
  node: true
extends:
  - plugin:react/recommended
  - plugin:import/typescript
  - prettier
parserOptions:
  ecmaVersion: latest
  sourceType: module
  ecmaFeatures:
    jsx: true
rules:
  eqeqeq:
    - error
  import/order:
    - error
    - alphabetize:
        order: asc
        caseInsensitive: true
parser: '@typescript-eslint/parser'
plugins:
  - react
overrides:
  - files:
      - '**/*.js'
    rules:
      '@typescript-eslint/no-var-requires':
        - 'off'
settings:
  import/parsers:
    '@typescript-eslint/parser':
      - .ts
      - .tsx
  react:
    version: detect
",
  "package.json": "{
  \\"name\\": \\"geofs-plugin\\",
  \\"version\\": \\"\\",
  \\"private\\": true
}",
  "src/App.js": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

import React from \\"react\\";

export function App() {
  // start writing your code here!
  // you can reference the geofs apis in intellisense because of the triple-slash directives
  // the apis from geofs plugin system are not available yet

  return <h1>Test</h1>;
}
",
  "src/index.js": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

import React from \\"react\\";
import { createRoot } from \\"react-dom/client\\";
import { App } from \\"./App\\";

// Replace the selector with a link to your UI host!
// Example: #geofs-ui-right
const root = createRoot(document.querySelector(\\"REPLACE ME\\"));
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
",
  "webpack.config.js": "const { join } = require(\\"node:path\\");

/** @type {import(\\"webpack\\").Configuration} */
const config = {
  entry: join(__dirname, \\"src\\", \\"index.js\\"),
  mode: \\"production\\",
  module: {
    rules: [
      {
        test: /\\\\.m?js$/,
        exclude: /node_modules/,
        use: {
          loader: \\"babel-loader\\",
          options: {
            presets: [\\"@babel/preset-env\\", \\"@babel/preset-react\\"],
          },
        },
      },
    ],
  },
  output: {
    clean: true,
    filename: \\"index.js\\",
    path: join(__dirname, \\"dist\\"),
  },
};

module.exports = config;
",
}
`;

exports[`integration tests integration test template:react-ts 1`] = `
Object {
  ".eslintrc.yml": "root: true
env:
  browser: true
  es2021: true
  node: true
extends:
  - plugin:react/recommended
  - plugin:import/typescript
  - prettier
parserOptions:
  ecmaVersion: latest
  sourceType: module
  ecmaFeatures:
    jsx: true
rules:
  eqeqeq:
    - error
  import/order:
    - error
    - alphabetize:
        order: asc
        caseInsensitive: true
parser: '@typescript-eslint/parser'
plugins:
  - react
overrides:
  - files:
      - '**/*.js'
    rules:
      '@typescript-eslint/no-var-requires':
        - 'off'
settings:
  import/parsers:
    '@typescript-eslint/parser':
      - .ts
      - .tsx
  react:
    version: detect
",
  "package.json": "{
  \\"name\\": \\"geofs-plugin\\",
  \\"version\\": \\"\\",
  \\"private\\": true
}",
  "src/App.tsx": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

import React from \\"react\\";

export function App() {
  // start writing your code here!
  // you can reference the geofs apis in intellisense because of the triple-slash directives
  // the apis from geofs plugin system are not available yet

  return <h1>Test</h1>;
}
",
  "src/index.tsx": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

import React from \\"react\\";
import { createRoot } from \\"react-dom/client\\";
import { App } from \\"./App\\";

// Replace the selector with a link to your UI host!
// Example: #geofs-ui-right
// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
const root = createRoot(document.querySelector(\\"REPLACE ME\\")!);
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
",
  "src/tsconfig.json": "{
  \\"compilerOptions\\": {
    \\"target\\": \\"es2015\\",
    \\"module\\": \\"esnext\\",
    \\"esModuleInterop\\": true,
    \\"moduleResolution\\": \\"node\\",
    \\"resolveJsonModule\\": true,
    \\"forceConsistentCasingInFileNames\\": true,
    \\"strict\\": true,
    \\"jsx\\": \\"react\\",
    \\"skipLibCheck\\": true
  },
  \\"include\\": [\\"**/*\\"]
}
",
  "webpack.config.js": "// @ts-check
const { join } = require(\\"node:path\\");

/** @type {import(\\"webpack\\").Configuration} */
const config = {
  entry: join(__dirname, \\"src\\", \\"index.js\\"),
  mode: \\"production\\",
  module: {
    rules: [
      {
        test: /\\\\.tsx?$/,
        exclude: /(node_modules)/,
        use: [
          {
            loader: \\"ts-loader\\",
          },
        ],
      },
    ],
  },
  output: {
    clean: true,
    filename: \\"index.js\\",
    path: join(__dirname, \\"dist\\"),
  },
};

module.exports = config;
",
}
`;

exports[`integration tests integration test template:typescript 1`] = `
Object {
  ".eslintrc.yml": "root: true
env:
  browser: true
  es2021: true
  node: true
extends:
  - plugin:@typescript-eslint/recommended
  - plugin:import/typescript
  - prettier
parserOptions:
  ecmaVersion: latest
  sourceType: module
rules:
  eqeqeq:
    - error
  import/order:
    - error
    - alphabetize:
        order: asc
        caseInsensitive: true
parser: '@typescript-eslint/parser'
plugins:
  - '@typescript-eslint'
overrides:
  - files:
      - '**/*.js'
    rules:
      '@typescript-eslint/no-var-requires':
        - 'off'
settings:
  import/parsers:
    '@typescript-eslint/parser':
      - .ts
      - .tsx
",
  "package.json": "{
  \\"name\\": \\"geofs-plugin\\",
  \\"version\\": \\"\\",
  \\"private\\": true
}",
  "src/index.ts": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

// start writing your code here!
// you can reference the geofs apis in intellisense because of the triple-slash directives
// the apis from geofs plugin system are not available yet
",
  "tsconfig.json": "{
  \\"compilerOptions\\": {
    \\"target\\": \\"es2015\\",
    \\"module\\": \\"esnext\\",
    \\"esModuleInterop\\": true,
    \\"moduleResolution\\": \\"node\\",
    \\"resolveJsonModule\\": true,
    \\"forceConsistentCasingInFileNames\\": true,
    \\"strict\\": true,
    \\"skipLibCheck\\": true
  },
  \\"include\\": [\\"**/*\\"]
}
",
  "webpack.config.js": "const { join } = require(\\"node:path\\");

/** @type {import(\\"webpack\\").Configuration} */
const config = {
  entry: join(__dirname, \\"src\\", \\"index.ts\\"),
  mode: \\"production\\",
  module: {
    rules: [
      {
        test: /\\\\.tsx?$/,
        exclude: /(node_modules)/,
        use: [
          {
            loader: \\"ts-loader\\",
          },
        ],
      },
    ],
  },
  resolve: {
    extensions: [\\".tsx\\", \\".ts\\", \\".js\\"],
  },
  output: {
    clean: true,
    filename: \\"index.js\\",
    path: join(__dirname, \\"dist\\"),
  },
};

module.exports = config;
",
}
`;

exports[`integration tests integration test vanilla 1`] = `
Object {
  ".eslintrc.yml": "root: true
env:
  browser: true
  es2021: true
  node: true
extends:
  - eslint:recommended
  - plugin:import/recommended
  - prettier
globals:
  PAGE_PATH: readonly
  geofs: readonly
  ui: readonly
  flight: readonly
  objects: readonly
  controls: readonly
  weather: readonly
  audio: readonly
  multiplayer: readonly
  instruments: readonly
  absMin: readonly
  span: readonly
  boundHours24: readonly
  fixAngle: readonly
  fixAngle360: readonly
  fixAngles360: readonly
  fixAngles: readonly
  exponentialSmoothingV3: readonly
  exponentialSmoothing: readonly
  getBuildingCollision: readonly
  xyz2lla: readonly
  xy2ll: readonly
  lla2xyz: readonly
  ll2xy: readonly
  clamp: readonly
  geoDecodeLocation: readonly
  lookAt: readonly
  getURLParameters: readonly
  clone: readonly
  PID: readonly
  Overlay: readonly
  rigidBody: readonly
  Object3D: readonly
  Indicator: readonly
  GlassPanel: readonly
parserOptions:
  ecmaVersion: latest
  sourceType: module
rules:
  eqeqeq:
    - error
  import/order:
    - error
    - alphabetize:
        order: asc
        caseInsensitive: true
",
  "package.json": "{
  \\"name\\": \\"geofs-plugin\\",
  \\"version\\": \\"\\",
  \\"private\\": true
}",
  "src/index.js": "/// <reference types=\\"jquery\\" />
/// <reference types=\\"@geps/geofs-types\\" />

// start writing your code here!
// you can reference the geofs apis in intellisense because of the triple-slash directives
// the apis from geofs plugin system are not available yet
",
  "webpack.config.js": "const { join } = require(\\"node:path\\");

/** @type {import(\\"webpack\\").Configuration} */
const config = {
  entry: join(__dirname, \\"src\\", \\"index.js\\"),
  mode: \\"production\\",
  module: {
    rules: [
      {
        test: /\\\\.m?js$/,
        exclude: /node_modules/,
        use: {
          loader: \\"babel-loader\\",
          options: {
            presets: [\\"@babel/preset-env\\"],
          },
        },
      },
    ],
  },
  output: {
    clean: true,
    filename: \\"index.js\\",
    path: join(__dirname, \\"dist\\"),
  },
};

module.exports = config;
",
}
`;
